src/windows.rs

```diff
@@ -235,7 +235,9 @@ fn stop_watch(ws: &WatchState, meta_tx: &Sender<MetaEvent>) {
         let ch = handleapi::CloseHandle(ws.dir_handle);
         // have to wait for it, otherwise we leak the memory allocated for there read request
         if cio != 0 && ch != 0 {
-            synchapi::WaitForSingleObjectEx(ws.complete_sem, INFINITE, TRUE);
+            while synchapi::WaitForSingleObjectEx(ws.complete_sem, INFINITE, TRUE) != WAIT_OBJECT_0
+            {
```

Comment:
If I'm correct this cold be a hot loop ? Because I'd rather sleep than let this run hot.

Reply:
I don't think it's a hot loop.

This loop will cause the thread to either handle the APC queue (in which case it will return WAIT_IO_COMPLETION), or once the APC which got queued from the CloseHandles above gets handled, will finish by meeting the WAIT_OBJECT_0 condition.

Sleeping will be unproductive since for the complete_sem to be in a signaled state, the APC that will ReleaseSemaphore will need to be called. And the only way to call and make progress on that front - is to WaitForSingleObjectEx to consume the APC queue.